### PR TITLE
[sca/capture] Add Ctrl-C handler to capture script

### PR DESCRIPTION
**Pull request by wettermo**
In case a huge number of traces is captured, this handler allows the
user, when capturing takes too much time, to abort the capture without
losing previously recorded data.
To use this handler it has to be registered within each of the capture
functions, where a Chipwhisperer project is used to store traces.

### CODE REVIEW

File: cw/capture.py [hunk 1]
```diff
@@ -17,12 +20,12 @@
...
 from Crypto.Hash import SHA3_256
-from cw_segmented import CwSegmented
-from pyXKCP import pyxkcp
 from tqdm import tqdm
-from waverunner import WaveRunner
 
+from cw_segmented import CwSegmented
+from pyXKCP import pyxkcp
 from util import device, plot
+from waverunner import WaveRunner
```
Comment (vogelpi) [inline cw/capture.py:26]: Question is this change needed due to lint or is it a remainder from the rebasing maybe?

File: cw/capture.py [hunk 2]
```diff
@@ -258,6 +274,7 @@ def capture_aes_random_batch(ot, ktp, capture_cfg, scope_type):
     num_segments_storage = 1
     # cw and waverunner scopes are supported fot batch capture.
     scope = SCOPE_FACTORY[scope_type](ot, capture_cfg)
+
```
Comment (vogelpi) [inline cw/capture.py:277]: I think this additioan newline (and the other two below) aren't required. I don't mind them being there to improve readability but then they should be added with the next commit which touches the surroundings of these lines anyway.

**Review by vogelpi — COMMENTED** [review]
Thanks @wettermo , this looks mostly good.

I have some nits, see comments and two more things:
1. Should we also register the handler inside the `capture_loop` function? This would also enable the feature in non-batch mode for AES, SHA3, KMAC.
2. The commit title of the first commit message is too long, it wraps.

File: cw/capture.py (hunk 1 above)
Comment (wettermo) [reply cw/capture.py:26]: I had a few problems with isort on my machine, which apparently kind of messed up the imports. I only realized that after I opened the PR, so I had to manually fix it.

File: cw/capture.py (hunk 2 above)
Comment (wettermo) [reply cw/capture.py:277]: I see, but I don't understand which next commit you mean.

Comment (wettermo) [reply cw/capture.py:277]: Ah sorry, got it!

**Comment by wettermo** [comment]
Thanks @vogelpi for the review!
Good point to also register the handler inside the capture_loop function. I'll do that and I'll also reforge the commits to get rid of the linting/isort issues.

**Comment by wettermo** [comment]
@vogelpi I cleaned up the commits, added the registering of the handler in the capture_loop, and put them together into one commit. Now it should look cleaner.

**Review by vogelpi — APPROVED** [review]
Thanks for addressing all comments @wettermo , this looks good to me!
